### PR TITLE
hdf5: Correct spelling

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 1]
```diff
@@ -79,12 +79,12 @@ class Hdf5(AutotoolsPackage):
...
     def libs(self):
-        """Hdf5 can be queried for the following parameters:
+        """HDF5 can be queried for the following parameters:
 
         - "hl": high-level interface
         - "cxx": C++ APIs
-        - "fortran": fortran APIs
-
+        - "fortran": Fortran APIs
+
```
Comment (davydden) [inline var/spack/repos/builtin/packages/hdf5/package.py]: white space

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hdf5/package.py]: I'm excited to see whether or not flake8 catches this. I recently fixed a bug that was letting trailing whitespace in.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hdf5/package.py]:
```
var/spack/repos/builtin/packages/hdf5/package.py:87: [W293] blank line contains whitespace
```
It feels so good...

Comment (eschnett) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Done.

**Pull request merged**